object/slicer/slicer.go

```diff
@@ -269,9 +261,16 @@ func initPayloadStream(ctx context.Context, ow ObjectWriter, header object.Objec
 
 	maxObjSize := childPayloadSizeLimit(opts)
 
-	res.buf.Grow(int(maxObjSize))
+	if uint64(len(opts.payloadBuffer)) > maxObjSize {
+		res.buf = opts.payloadBuffer[:maxObjSize+1]
```

Comment:
It worked with `maxObjSize` previously, do we need `+1`? It's very inconvenient for API users.

Reply:
agree, this is not very convenient. But this additional byte allows slicer to work using single payload buffer. Idk why, but previously slicer used two buffers - one external and one internal of `maxObjSize`. It _can_ work with single buffer. But for this we need additional byte in order to distinguish "small" objects from "big" ones cuz they are formed completely different

if we'll find the way how to get by with `maxObjSize` buffer, then we'll lighten the requirement and this will be backward compatible. Again, one byte is nothing

from other side, i completely dont like preset buffers cuz slicer could need _bigger_ buffer in some future, so some code could become insta useless. Slicer can also stop (fail) before it actually use the buffer - we get wasted allocation. This is not how it's usually done. But this one is #544

P.S. i really wanna implement #544 and drop `SetPayloadBuffer` in this PR, any objections?

Reply:
Technically you can handle this byte internally, it'd just cost a single additional `Read()` for the first chunk.

Reply:
`+1` is still a problem, people will make mistakes with it.